### PR TITLE
add function exist condition

### PR DESCRIPTION
**Pull request by Pradeep1308**
**What Changed? :hammer:** 

Add `function_exist()` condition check

ref: #1798

### CODE REVIEW

**Comment by Pradeep1308** [comment]
@pavanpatil1 check this PR

**Comment by pavanpatil1** [comment]
@Pradeep1308  Cross-checked the PR, fix is working fine the message is not displayed in the log. 

https://user-images.githubusercontent.com/44057535/147028876-c08a2112-dae2-47a3-b7a6-1dfbb8ad1812.mp4

But there is another regression issue. After enabling the group, the fatal error for `function_exits()` is displayed on the group media screen. 

Steps: Enable the user's group. From subscriber users' accounts, go to all media and observe the log. 

image

**Comment by Pradeep1308** [comment]
@pavanpatil1  check this again. :+1:

**Comment by pavanpatil1** [comment]
Working fine now. You can merge the PR @Pradeep1308.

File: app/main/controllers/template/rtmedia-functions.php [hunk 1]
```diff
@@ -1039,7 +1039,21 @@ function rtmedia_edit_allowed() {
...
+		if ( ! $flag ) {
+			$flag = bp_group_is_admin();
+		}
+	}
+
+	if ( function_exists( 'bp_group_is_mod' ) ) {
+
+		if ( ! $flag ) {
+			$flag = bp_group_is_mod();
+		}
```
Comment (sabbir1991) [inline app/main/controllers/template/rtmedia-functions.php:1056]: Can we make a separate function for this function_exist checking. Like
```
function rtm_is_bp_group_admin() {
   return function_exists( 'bp_group_is_admin' ) && bp_group_is_admin();
}
```

Comment (Pradeep1308) [reply app/main/controllers/template/rtmedia-functions.php:1056]: yes we can do it, I"ll also add separate function for moderator.

File: app/main/controllers/template/rtmedia-filters.php [hunk 2]
```diff
@@ -81,7 +81,9 @@ function rtmedia_album_edit( $options ) {
 
 	if ( isset( $rtmedia_query->media_query ) && isset( $rtmedia_query->media_query['album_id'] ) && ! in_array( intval( $rtmedia_query->media_query['album_id'] ), array_map( 'intval', rtmedia_get_site_option( 'rtmedia-global-albums' ) ), true ) ) {
 
-		if ( rtmedia_is_album_editable() || is_rt_admin() || bp_group_is_admin() || bp_group_is_mod() ) {
+		if ( rtmedia_is_album_editable() || is_rt_admin()
+		|| ( function_exists( 'bp_group_is_admin' ) && bp_group_is_admin() )
+		|| ( function_exists( 'bp_group_is_mod' ) && bp_group_is_mod() ) ) {
```
Comment (sabbir1991) [inline app/main/controllers/template/rtmedia-filters.php:86]: You need to remove this codes. and use the custom function you have created.

File: app/main/controllers/template/rtmedia-functions.php [hunk 3]
```diff
@@ -2996,6 +2996,25 @@ function is_rt_admin() {
 
 }
 
+/**
+ * Checking if is group admin.
+ *
+ * @return      bool
```
Comment (sabbir1991) [inline app/main/controllers/template/rtmedia-functions.php:3002]: Remove extra space from doc comment

File: app/main/controllers/template/RTMediaTemplate.php [hunk 4]
```diff
@@ -1066,7 +1066,9 @@ public static function locate_template( $template = false, $context = false, $ur
 
 				if ( is_rtmedia_album() && isset( $rtmedia_query->media_query ) && 'edit' === $rtmedia_query->action_query->action
 				) {
-					if ( rtmedia_is_album_editable() || is_rt_admin() || bp_group_is_admin() || bp_group_is_mod() ) {
+					if ( rtmedia_is_album_editable() || is_rt_admin()
+					|| ( function_exists( 'bp_group_is_admin' ) && bp_group_is_admin() )
+					|| ( function_exists( 'bp_group_is_mod' ) && bp_group_is_mod() ) ) {
```
Comment (sabbir1991) [inline app/main/controllers/template/RTMediaTemplate.php:1071]: here also need to add you custom function check

**Review by sabbir1991 — APPROVED** [review]

**Pull request merged**